keanu-project/src/main/java/io/improbable/keanu/network/BayesianNetwork.java

```diff
@@ -18,15 +15,34 @@
 public class BayesianNetwork {
 
     private final List<? extends Vertex> vertices;
+    private final HashMap<String, Vertex> labelSet;
```

Comment:
Since you don't often need to know the exact implementation of the Map, we define it to the lowest class that still satisfies our needs. In this case it's just `private final Map<String, Vertex> labelSet;`